### PR TITLE
[daemon] Fix mapping of instance/snapshot args

### PR DESCRIPTION
**Pull request by ricab**
Needs #3092

### CODE REVIEW

File: src/daemon/daemon.cpp [hunk 1]
```diff
@@ -1168,7 +1168,7 @@ std::unordered_map<std::string, std::unordered_set<std::string>> map_snapshots_t
     for (const auto& it : instances)
     {
         if (it.snapshot_name().empty())
-            instance_snapshots_map[it.instance_name()];
+            instance_snapshots_map[it.instance_name()].clear();
```
Comment (ricab) [inline src/daemon/daemon.cpp:1171]: Using clear to make it obvious that we're removing anything that is there already.

**Comment by ricab** [comment]
I did this on top of your delete branch to avoid having to deal with conflicts there.

**Review by sharder996 — APPROVED** [review]
LGTM!

**Comment by ricab** [comment]
Thanks for the review @sharder996. Did a clean rebase and merging.

**Pull request merged**